python/lsst/ts/salobj/base_script.py

```diff
@@ -214,49 +208,72 @@ def make_from_cmd_line(
             "The index argument is ignored, though it is still required.",
         )
         args = parser.parse_args()
-        kwargs = dict(index=args.index)
-        if descr is not None:
-            kwargs["descr"] = descr
+
         if args.schema:
             schema = cls.get_schema()
             if schema is not None:
                 print(yaml.safe_dump(schema))
             return None
-        return cls(**kwargs)
+
+        return cls(index=args.index, **kwargs)
 
     @classmethod
-    async def amain(cls, descr: typing.Optional[str] = None) -> None:
+    async def amain(cls, **kwargs: typing.Any) -> None:
         """Run the script from the command line.
 
         Parameters
         ----------
-        descr : `str`, optional
-            Short description of what the script does, for operator display.
-            Leave at None if the script already has a description, which is
-            the most common case. Primarily intended for unit tests,
-            e.g. running ``TestScript``.
-
+        kwargs :
+            Keyword arguments for the script constructor.
+            Must not include ``index``.
+            Ignored (other than testing for ``index``)
+            if the command specifies ``--schema``.
 
         Notes
         -----
         The final return code will be:
 
         * 0 if final state is `lsst.ts.idl.enums.Script.ScriptState.DONE`
```

Comment:
These return codes look like ideal candidates to extract into an enum. Maybe for a future ticket?

Reply:
I am actually leaning towards removing all these special return codes and simply returning 1. The ScriptQueue is not using them and I can't think of a use for them. The script reports its final state using SAL.

Reply:
I decided to simplify exit codes to 0/1 after looking at ts_scriptqueue and finding no usage of the exit code. All unit tests pass in ts_scriptqueue.